futures-core/src/task/mod.rs

```diff
@@ -120,10 +121,38 @@ if_std! {
         }
     }
 
+    #[cfg(not(feature = "nightly"))]
+    unsafe impl<F: Future<Output = ()> + Send + 'static + ::Unpin> UnsafePoll for Box<F> {
```

Comment:
Why the `Unpin` bound? You're boxing the value, so you know that it's safe to be polled, and it can't ever be moved out, right?

Reply:
When not on nightly, you *have* to have `Unpin` because the only method you can call on a future requires it.

Reply:
Right, because the other method doesn't exist. Thanks!